app_dart/lib/src/service/gerrit_service.dart

```diff
@@ -0,0 +1,33 @@
+// Copyright 2020 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert' as convert;
+
+import 'package:http/http.dart' as http;
+
+import 'logging.dart';
+
+class GerritService {
+  GerritService({http.Client? httpClient}) : httpClient = httpClient ?? http.Client();
+
+  final http.Client httpClient;
+
+  /// Gets the branches from a remote git repository using the gerrit APIs.
+  Future<List<String>> branches(String repo, String project, String prefix) async {
+    final Uri url = Uri.https(repo, 'projects/$project/branches', <String, dynamic>{'m': prefix});
+    final http.Response response = await httpClient.get(url);
```

Comment:
Can we use existing http get functions defined in `untils.dart` which has the retry logic?

Reply:
I believe you are referring to githubFileContent or getBranches which seem very specific to github. I could not find generic http implementations.

Reply:
can we add a retry logic to the http request to bypass some transient http errors? this seems being called on every single pre-submit test.

Reply:
This is only executed once per PR, added retries.